### PR TITLE
Fix error bars on plot_excess method of SpectrumDataset and SpectrumDatasetOnOff and plot_residuals_spectral

### PR DESCRIPTION
**Pull request by mdebony**
**Description**

This pull request is a for solving issue  #3971 about the incorrect size of the error bars on plot created by the method plot_excess of SpectrumDataset and SpectrumDatasetOnOff. This also add correct error bar computation for all the method of the plot_residuals_spectral method used by all Datasets.

**Dear reviewer**
This is pull request is ready for review.

### CODE REVIEW

**Review by registerrier — CHANGES_REQUESTED** [review]
Thanks @mdebony .

Since this is a bug fix release correction, we should not change the API so the addition of the 'diff/model' option should be left out. 
It would be good to check the impact of the modified 'diff/sqrt(model)' option. How does it impact the resulting plots?

File: gammapy/datasets/map.py [hunk 1]
```diff
@@ -953,25 +953,29 @@ def plot_residuals_spectral(self, ax=None, method="diff", region=None, **kwargs)
...
-                    n_off=counts_off,
-                    alpha=alpha,
-                    mu_sig=mu_sig,
-                )
-            elif self.stat_type == "cash":
-                stat = CashCountsStatistic(counts_spec.data, npred_spec.data)
-            yerr = stat.error
+            yerr = excess_error
+        elif method == "diff/model":
+            yerr = excess_error/npred_spec.data
```
[the remark this answers is not in the document]
Comment (mdebony) [reply gammapy/datasets/map.py:976]: Ok, as the method is available on the method _compute_residuals which is being used for the plot function, I thought it was worth adding it.

File: gammapy/datasets/map.py [hunk 2]
```diff
@@ -953,25 +953,29 @@ def plot_residuals_spectral(self, ax=None, method="diff", region=None, **kwargs)
...
         elif method == "diff/sqrt(model)":
-            yerr = np.ones_like(residuals.data)
+            yerr = excess_error/np.sqrt(npred_spec.data)
```
[the remark this answers is not in the document]
Comment (mdebony) [reply gammapy/datasets/map.py:978]: The ones likes was just defaulting the errors bars to one so they weren't really corresponding to anything.
I've added a plot with the two error bars over imposed to show the difference (test base on the tutorial notebook for spectral analysis).
image

**Comment by adonath** [comment]
Thanks @mdebony! We should leave a description in the docstring what it actually plots now...to avoid confusion in future.

**Comment by adonath** [comment]
Thanks @mdebony! We should leave a description in the docstring what it actually plots now...to avoid confusion in future.

**Comment by mdebony** [comment]
Ok I've added a precision on docstrings of both methods about the way of the error bars are computed. Was it what you were expected ?

File: gammapy/datasets/map.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (registerrier) [reply gammapy/datasets/map.py:978]: OK.I assume that the smaller error bars at high energy are due to the fact that the excess error can become null.

**Review by registerrier — APPROVED** [review]
Thanks @mdebony !

**Pull request merged**